src/rc_allocator.c

```diff
@@ -441,9 +440,14 @@ rc_allocator_mount(rc_allocator        *al,
 
    uint32 buffer_size = cfg->extent_capacity * sizeof(uint8);
    buffer_size        = ROUNDUP(buffer_size, cfg->io_cfg->page_size);
-   al->bh = platform_buffer_create(buffer_size, al->heap_handle, mid);
-   platform_assert(al->bh != NULL);
-   al->ref_count = platform_buffer_getaddr(al->bh);
+   status             = platform_buffer_init(&al->bh, buffer_size);
+   if (!SUCCESS(status)) {
+      platform_mutex_destroy(&al->lock);
+      platform_free(al->heap_id, al->meta_page);
```

Comment:
Although technically, if you want to undo allocations in the reverse order in which they were done, I believe the `platform_mutex_destroy` should come after the `platform_free`.

Reply:
Yes, right.  Changed it accordingly.